### ISSUE TITLE
[accessibility]: accessibility issue with combox

### ISSUE THREAD

**Issue by cedricpelle** [carbon-design-system/carbon-for-ibm-dotcom#11268]
### Description

i have some violations when using the IBM Equal Access Accessibility Checker on the combox
![image](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/38075847/53923603-3536-49c2-a945-b1a85cf81d0f)

[Accessibility_Report-test.xlsx](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/files/13751465/Accessibility_Report-test.xlsx)

### Component(s) impacted

combox

### Carbon for IBM.com version

V1

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

IBM Terms

### Package

@carbon/ibmdotcom-web-components, @carbon/web-components

### CodeSandbox example

https://web-components-v1.carbondesignsystem.com/?path=/story/components-combo-box--default

### Steps to reproduce the issue (if applicable)

Code used :

<link rel="stylesheet" href="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/version/v1.37.0/plex.css" />
    <link rel="stylesheet" href="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/version/v1.37.0/grid.css" />
    <link rel="stylesheet" href="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/version/v1.37.0/themes.css" />
    

    <script type="module"
    src="https://1.www.s81c.com/common/carbon/web-components/version/v1.35.0/combo-box.min.js"></script>

<bx-combo-box
  helper-text="Optional helper text"
  label-text="Combo box title"
  trigger-content="Filter..."
  aria-label="Combo box"
>
  <bx-combo-box-item value="all">Option 1</bx-combo-box-item>
  <bx-combo-box-item value="cloudFoundry">Option 2</bx-combo-box-item>
  <bx-combo-box-item value="staging">Option 3</bx-combo-box-item>
  <bx-combo-box-item value="dea">Option 4</bx-combo-box-item>
  <bx-combo-box-item value="router">Option 5</bx-combo-box-item>
  <bx-combo-box-item value="support">Option 6</bx-combo-box-item>
  <bx-combo-box-item value="services">Option 7</bx-combo-box-item>
  <bx-combo-box-item value="products">Option 8</bx-combo-box-item>
</bx-combo-box> 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by cedricpelle** [comment]
it is the code used

`  <bx-combo-box
  helper-text="Optional helper text"
  label-text="Combo box title"
  trigger-content="Filter..."
  aria-label="Combo box"
>
  <bx-combo-box-item value="all">Option 1</bx-combo-box-item>
  <bx-combo-box-item value="cloudFoundry">Option 2</bx-combo-box-item>
  <bx-combo-box-item value="staging">Option 3</bx-combo-box-item>
  <bx-combo-box-item value="dea">Option 4</bx-combo-box-item>
  <bx-combo-box-item value="router">Option 5</bx-combo-box-item>
  <bx-combo-box-item value="support">Option 6</bx-combo-box-item>
  <bx-combo-box-item value="services">Option 7</bx-combo-box-item>
  <bx-combo-box-item value="products">Option 8</bx-combo-box-item>
</bx-combo-box>`

**Comment by cedricpelle** [comment]
code used

**Comment by cedricpelle** [comment]
code used 
![image](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/38075847/a434cf5d-3bd2-4a1f-bc5e-5be594e84298)

**Comment by cedricpelle** [comment]
We have too a violation on the button to unselect 

![image](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/38075847/0ace34dc-8ea0-4f2c-a844-a434338258ea)
